src/MoBi.Core/Commands/SetParentPathCommand.cs

```diff
@@ -30,9 +32,46 @@ protected override void ExecuteWith(IMoBiContext context)
       {
          base.ExecuteWith(context);
          _container.ParentPath = _newParentPath;
+         updateObjectPathsInSpatialStructure();
          Description = AppConstants.Commands.UpdateParentPath(_container.EntityPath(), (_newParentPath ?? new ObjectPath()).ToPathString());
       }
 
+      private void updateObjectPathsInSpatialStructure()
+      {
+         //should probably never be null
+         var spatialStructure = _buildingBlock as SpatialStructure;
+         if (spatialStructure == null)
+            return;
+
+         var oldObjectPathString = createContainerPath(_oldParentPath);
+         var newObjectPathString = createContainerPath(_newParentPath);
```

Comment:
would rename the variables to `oldParentContainerPathString `and `newParentContainerPathString `for clarity

Reply:
check that the old parent path was not empty. If it was - nothing to do?